src/main/flight/interpolated_setpoint.c

```diff
@@ -70,112 +67,86 @@ void interpolatedSpInit(const pidProfile_t *pidProfile) {
-        // determine if this step was a relatively large one, to use when evaluating next packet
+        float absSetpointSpeed = fabsf(setpointSpeed); // unsmoothed for kick prevention
 
-        if (absSetpointSpeed > 1.5f * absPrevSetpointSpeed || absPrevSetpointSpeed > 1.5f * absSetpointSpeed){
-            bigStep[axis] = true;
-        } else {
-            bigStep[axis] = false;
-        }
+        // calculate acceleration, smooth and attenuate it
+        setpointAcceleration = setpointSpeed - prevSetpointSpeed[axis];
+        setpointAcceleration = prevAcceleration[axis] + ffSmoothFactor * (setpointAcceleration - prevAcceleration[axis]);
+        setpointAcceleration *= ffAttenuator;
 
-        // smooth deadband type suppression of FF jitter when sticks are at or returning to centre
-        // only when ff_averaging is 3 or more, for HD or cinematic flying
-        if (averagingCount > 2) {
-            const float rawSetpointCentred = absRawSetpoint / averagingCount;
-            if (rawSetpointCentred < 1.0f) {
-                setpointSpeed *= rawSetpointCentred;
-                setpointAcceleration *= rawSetpointCentred;
-            }
-        }
+        // smooth setpointSpeed but don't attenuate
+        setpointSpeed = prevSetpointSpeed[axis] + ffSmoothFactor * (setpointSpeed - prevSetpointSpeed[axis]);
 
+        prevSetpointSpeed[axis] = setpointSpeed;
         prevAcceleration[axis] = setpointAcceleration;
-     
-        // all values afterwards are small numbers
+        prevRcCommandDelta[axis] = rcCommandDelta;
+
         setpointAcceleration *= pidGetDT();
         setpointDeltaImpl[axis] = setpointSpeed * pidGetDT();
 
-
+        // calculate boost and prevent kick-back spike at max deflection
         const float ffBoostFactor = pidGetFfBoostFactor();
         float boostAmount = 0.0f;
-        if (ffBoostFactor != 0.0f) {
-            // calculate boost and prevent kick-back spike at max deflection
-            if (fabsf(rawSetpoint) < 0.95f * ffMaxRate[axis] || absSetpointSpeed > 3.0f * absPrevSetpointSpeed) {
-                boostAmount = ffBoostFactor * setpointAcceleration * setpointAccelerationModifier;
+        if (ffBoostFactor) {
+            if (fabsf(setpoint) < 0.95f * ffMaxRate[axis] || absSetpointSpeed > 3.0f * absPrevSetpointSpeed) {
+                boostAmount = ffBoostFactor * setpointAcceleration;
             }
         }
 
-        prevSetpointSpeed[axis] = setpointSpeed;
-        prevRawSetpoint[axis] = rawSetpoint;
-
         if (axis == FD_ROLL) {
-            DEBUG_SET(DEBUG_FF_INTERPOLATED, 0, lrintf(setpointDeltaImpl[axis] * 100));
-            DEBUG_SET(DEBUG_FF_INTERPOLATED, 1, lrintf(setpointAcceleration * 100));
-            DEBUG_SET(DEBUG_FF_INTERPOLATED, 2, lrintf(((setpointDeltaImpl[axis] + boostAmount) * 100)));
-            DEBUG_SET(DEBUG_FF_INTERPOLATED, 3, ffStatus[axis]);
+            DEBUG_SET(DEBUG_FF_INTERPOLATED, 0, lrint(setpointDeltaImpl[axis] * 100.0f)); // base FF
```

Comment:
Should be `lrintf()` - `lrint()` expects double precision float as input.

Reply:
good catch!  I missed that.  Thanks!